### ISSUE TITLE
Package dependencies and non-dependencies in desisim

### ISSUE THREAD

**Issue by weaverba137** [desihub/desisim#43]
While analyzing the packages that desisim depends on, I found a few anomalies that I need to ask about.

1. Why *doesn't* desisim depend on specsim?
2. Is matplotlib strictly necessary for the functioning of the package?
3. Why does code in the qso_template section depend on `xastropy`?

**Comment by weaverba137** [comment]
PS, the purpose of this question is to determine exactly which packages need to be installed for Travis tests to succeed.

**Comment by sbailey** [comment]
> Why doesn't desisim depend on specsim?

The parts of desisim that would depend on specsim (quickgen) pre-dated the existence of specsim (though the final merge did not).  In the future it will depend on specsim, but we just haven't done the refactor yet.

> Is matplotlib strictly necessary for the functioning of the package?

It is needed for the QA plots but not the algorithms.  See comment on [desispec #94](https://github.com/desihub/desispec/issues/94) for skipping tests if matplotlib is a limiting factor for Travis.

> Why does code in the qso_template section depend on xastropy?

@profxj was probably being pragmatic in using the tools he had available.  He can comment on how easy it would be to clean that up when merging qso_template.

**Comment by weaverba137** [comment]
What is the schedule for merging qso_template into the main desisim package?  I'm waiting on that to happen so I can determine the final set of dependencies.

**Comment by dkirkby** [comment]
Are you waiting on the quicksim -> specsim migration, or can we finalize the current dependencies to close this issue and then update them as part of #44 ?

**Comment by weaverba137** [comment]
Yes, I'm waiting for the migration.  We should have only one package in the py/ directory.

**Comment by dkirkby** [comment]
The quicksim -> specsim migration is unrelated to merging qso_template into the main desisim package.

**Comment by weaverba137** [comment]
Sorry, by 'migration', I specifically meant the merge of qso_template.

**Comment by sbailey** [comment]
qso_template has been merged into desisim/ (PR #67) and the quicksim -> specsim migration is complete (#57 and #86).  Closing ticket.
